### DISCUSSION TITLE
Cannot connect to SIMATIC S7-1500

### DISCUSSION THREAD

**Discussion by MarcelloMagni** [eclipse-milo/milo#1165] [General]
As title says,
i try to create the following  connection:

`OpcUaClient client = OpcUaClient.create(getUrlConnection(), endpoints -> endpoints.stream()
								.filter(e -> e.getSecurityPolicyUri().equals(SecurityPolicy.None.getUri())).findFirst(),
								configBuilder -> configBuilder.build());
UaClient clientFuture = client.connect().get();`

but the following execution exception is thown:

status=Bad_Timeout, message=io.netty.channel.ConnectTimeoutException: connection timed out

The machine uses a NAT and the manufacturer suggests me to change the URL: [https://support.industry.siemens.com/cs/document/109766709/what-are-the-causes-when-connection-to-an-opc-ua-server-fails-?dti=0&lc=en-HK](url)

How can i achieve this? Isn't the connection enstablished in the correct way? 
The getUrlConnection() method returns the local IP and the Exception is thrown on another IP.

**Comment by kevinherron** [comment]
If the server responds with endpoints that contain hostnames and/or ports that you cannot resolve or reach as a client you can replace them as described in this post: https://stackoverflow.com/a/40555243/85927

**Reply by MarcelloMagni** [reply]
Thank you very much, using: 

`EndpointUtil.updateUrl`

I was able to establish the connection.
Sorry for not having found that post earlier.
